Gems/SimulationInterfaces/Code/Source/Clients/NamedPosesManager.cpp

```diff
@@ -0,0 +1,291 @@
+    {
+        dependent.push_back(AZ_CRC_CE("PhysicsService"));
+        dependent.push_back(AZ_CRC_CE("SimulationFeaturesAggregator"));
+    }
+
+    NamedPoseManager::NamedPoseManager()
+    {
+        if (NamedPoseManagerRequestBusInterface::Get() == nullptr)
+        {
+            NamedPoseManagerRequestBusInterface::Register(this);
+        }
+    }
+
+    NamedPoseManager::~NamedPoseManager()
+    {
+        if (NamedPoseManagerRequestBusInterface::Get() == this)
+        {
+            NamedPoseManagerRequestBusInterface::Unregister(this);
+        }
+    }
+
+    void NamedPoseManager::Init()
+    {
+    }
+
+    void NamedPoseManager::Activate()
+    {
+        SimulationFeaturesAggregatorRequestBus::Broadcast(
+            &SimulationFeaturesAggregatorRequests::AddSimulationFeatures,
+            AZStd::unordered_set<SimulationFeatureType>{
+                simulation_interfaces::msg::SimulatorFeatures::NAMED_POSES,
+                simulation_interfaces::msg::SimulatorFeatures::POSE_BOUNDS,
+            });
+
+        NamedPoseManagerRequestBus::Handler::BusConnect();
+    }
+
+    void NamedPoseManager::Deactivate()
+    {
+        NamedPoseManagerRequestBus::Handler::BusDisconnect();
+    }
+
+    AZ::Outcome<void, FailedResult> NamedPoseManager::RegisterNamedPose(AZ::EntityId namedPoseEntityId)
+    {
+        // check if given entity has named pose component
+        AZ::Entity* entity = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationBus::Events::FindEntity, namedPoseEntityId);
+        if (!entity)
+        {
+            return AZ::Failure(
+                FailedResult(simulation_interfaces::msg::Result::RESULT_OPERATION_FAILED, "Failed to gather entity with given id"));
+        }
+        auto namedPoseComponent = entity->FindComponent<NamedPoseComponent>();
+        if (!namedPoseComponent)
+        {
+            return AZ::Failure(FailedResult(
+                simulation_interfaces::msg::Result::RESULT_OPERATION_FAILED,
+                "Entity with given Id doesn't have NamedPoseComponent, use CreateNamedPose method to create new pose"));
+        }
+
+        NamedPose configuration;
+        NamedPoseComponentRequestBus::EventResult(configuration, namedPoseEntityId, &NamedPoseComponentRequests::GetConfiguration);
+        if (m_namedPoseToEntityId.contains(configuration.m_name))
+        {
+            return AZ::Failure(FailedResult(
+                simulation_interfaces::msg::Result::RESULT_OPERATION_FAILED,
+                "Named pose with given name already exists in the record. Change name of your entity with NamedPoseComponent"));
+        }
+        m_namedPoseToEntityId[configuration.m_name] = namedPoseEntityId;
+        return AZ::Success();
+    }
+
+    AZ::Outcome<void, FailedResult> NamedPoseManager::UnregisterNamedPose(AZ::EntityId namedPoseEntityId)
```

Comment:
```suggestion
    AZ::Outcome<void, FailedResult> NamedPoseManager::UnregisterNamedPose(AZ::EntityId& namedPoseEntityId)
```
Is there a reason to do not use a reference?